src/main.c

```diff
@@ -187,6 +186,7 @@ int main(int argc, char **argv) {
   if (lua_toboolean(L, -1)) {
     lua_close(L);
     rencache_invalidate();
+    dirmonitor_deinit();
```

Comment:
This causes core dumps. I don't think it's supposed to be re-initialized multiple times.
We should just iterate over the watch ids we know and unwatch them.

Reply:
Hrm. That's kind of irritating. Ideally, obviously, dmon should support this. I agree that's definitely the quickest way to watch/unwatch; but we should probably also submit a bug report. Ideally initing/deiniting should be idempotent.

Reply:
I've reverted this back to what it was, for now.